### PR TITLE
docs: add TCP guide

### PR DESCRIPTION
**Pull request by travisgroth**
## Summary
Add example config and guide for TCP support and SSH helper support

## Related issues
#1712 

**Checklist**:
- [x] add related issues
- [x] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [x] ready for review

### CODE REVIEW

**Review by desimone — APPROVED** [review]
Looks great!

File: docs/guides/tcp.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+The following guide demonstrates how to use Pomerium's [TCP Proxying](/topics/tcp-support.md) support with various TCP services such as databases and other non-HTTP protocols.  It also covers integration points with them when possible.
+
+The source files from this guide can be found on [GitHub](https://github.com/pomerium/pomerium/tree/master/examples/tcp/).
+
+## Background
+
+Pomerium is primarily designed to protect HTTP based services but, when replacing a traditional VPN, there are often non-HTTP based applications which must still be reachable.  In order to provide the same type of protection to these services, Pomerium supports proxying TCP connections to upstream servers with the help of a client side application.  Authentication and authorization configuration is shared with standard HTTP routes, and the underlying transport is still encrypted between the end-user and Pomerium.
+
+It is worth noting that Pomerium authorizes HTTP on a request-by-request basis, but TCP applications are authorized on a per-connection basis.  At this time, Pomerium does not have integrations with providing credentials to upstream applications.
```
[the remark this answers is not in the document]
Comment (travisgroth) [reply docs/guides/tcp.md:21]: rephrased this section a bit

File: docs/guides/tcp.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+## How it works
```
[the remark this answers is not in the document]
Comment (travisgroth) [reply docs/guides/tcp.md:26]: Agreed.  I was thinking the same but would like to do as a follow up so the guide isn't blocked.

**Review by desimone — APPROVED** [review]
LGTM!

**Pull request merged**